### PR TITLE
Support for @Health annotations

### PR DESCRIPTION
**Pull request by heiko-braun**
Includes support for security realms

NOTE: Requires https://github.com/wildfly-swarm/wildfly-swarm-spi/pull/1 to be merged first

### CODE REVIEW

File: container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java [hunk 1]
```diff
@@ -69,13 +71,31 @@ public void debug(boolean debug) {
...
+            throw new RuntimeException(e);
+        }
+
+        Index index = indexer.complete();
+
+
+        // 2. give the factions a chance to handle the deployment
         for (ServerConfiguration each : this.configurations) {
             each.prepareArchive(deployment);
+            each.processMetaData(deployment, index);
```
Comment (kenfinnigan) [inline container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java]: We may want to do the metadata processing in a loop after this one, to account for one Fraction doing something in prepareArchive() that impacts another Fractions processMetaData().

Thinking some more, we may even want to create the meta data index after looping through prepareArchive()

**Comment by kenfinnigan** [comment]
Does the spi module.xml in /container need to be updated to have a dependency on jandex module? Now that the spi has a provided dependency for it?

**Comment by heiko-braun** [comment]
The last commit adds the security realm support

**Comment by kenfinnigan** [comment]
With the latest changes, is the MapIdentityManager needed still?

**Comment by heiko-braun** [comment]
Yes, the Map IDM is not needed anymore

**Comment by heiko-braun** [comment]
@kenfinnigan please keep this on hold. I still need to tweak the authentication bits a little bit

File: container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java [hunk 2]
```diff
@@ -69,13 +71,34 @@ public void debug(boolean debug) {
     @Override
     public void deploy(Archive<?> deployment) throws DeploymentException {
 
+        // 1. give the factions a chance to handle the deployment
```
Comment (kenfinnigan) [inline container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java]: Apologies Heiko, you're going to be annoyed by this! After saying what I did on Friday, I think, I realized that maybe some calls to prepareArchive() need to rely on things in the index? I'm not entirely sure, but it was something that popped into my head.

For instance, would we want prepareArchive() to be able to search for annotations, etc to then add/remove code from the Archive depending on what they find?

If so, then we do need to index first, and now you hate me ;-)

So, if we took that approach, do we need to re-index the index after prepareArchive()?

**Comment by kenfinnigan** [comment]
No problem, wasn't planning to merge just yet

File: container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java (hunk 2 above)
Comment (heiko-braun) [inline container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java]: I'd suggest we'll leave it as is until we hit the cases you mention.

**Comment by heiko-braun** [comment]
Cleaned up and squashed everything into one commit. From my point of view this is good to go. I'd suggest to move any further into dedicated branches (i.e. @kenfinnigan thoughts on the RuntimeDeployer)

**Comment by heiko-braun** [comment]
My vote on merging this one: +1 

@kenfinnigan ?

File: container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java (hunk 2 above)
Comment (gastaldi) [inline container/runtime/src/main/java/org/wildfly/swarm/container/runtime/RuntimeDeployer.java]: s/factions/fractions

**Comment by heiko-braun** [comment]
The latest commit fixes a typo and squashes everything into one again

**Comment by gastaldi** [comment]
Looks good :+1:

**Comment by kenfinnigan** [comment]
Just a quickie on RuntimeDeployer. Would we envisage that something like https://github.com/wildfly-swarm/wildfly-swarm-swagger/blob/master/adapter/runtime/src/main/java/org/wildfly/swarm/swagger/runtime/SwaggerConfiguration.java#L63 is replaced with using the index or not?

**Comment by heiko-braun** [comment]
Looks like I broke the security with one of my latest commits. I'll need to enhance the test coverage for this and fix it, before we can move on.

**Comment by heiko-braun** [comment]
@kenfinnigan re: Swagger. Yes, that's one of the intended use cases.

**Comment by kenfinnigan** [comment]
Given that, we will need to prepare the index before calling prepareArchive() but also pass the index into that method, which is an SPI change.

Is there an alternative I haven't thought of?

**Comment by heiko-braun** [comment]
@kenfinnigan we can still shuffle things around later, but I would suggest not to make all possible use cases (i.e. swagger) a pre-requisite for merging what we have so far.

**Comment by heiko-braun** [comment]
Squashed everything again and fixed the JAXRSArquillianTest

**Pull request merged**